src/main/java/formflow/library/ScreenController.java

```diff
@@ -237,6 +234,11 @@ ModelAndView getSubflowScreen(
     return new ModelAndView(String.format("%s/%s", flow, screen), model);
   }
 
+  @NotNull
+  private static ModelAndView errorScreen() {
+    return new ModelAndView(new RedirectView("/error"));
```

Comment:
Are we able to pass more context into this, like the "BAD_REQUEST" info? It seems like we are losing that information when generating the error page.

Reply:
I can add it back in but I wasn't sure what we we're getting by reporting it. If we have the the full trace on we'd see all the relevant information, if not we'd just get redirected to the client facing error page. Am I missing something?

Reply:
In the case that the user went to a page in the flow that didn't exist `/flow/ubi/junkPage`, the error page didn't report anything useful and had a 999 error (image above).   There was no stack trace either.  Mostly, it'd be good to have better error reporting in that scenario. So, not sure if passing in more context would work, but definitely having more messaging in that one scenario would be good.

Reply:
So the approach of using `redirect:` was not allowing status code or anything else to get passed to the `/error` controller mapping.

I started a [WIP approach of creating a model of info](https://github.com/codeforamerica/form-flow/pull/231/files#diff-e3c0b2bc250fdece31f72615351249643e60c4ce9dff6cad2730fb9ea87dc9efR240-R251) needed to pass to the error page but this approach essentially repeats much of the same code that lives in `FormFlowErrorController` 😢 

Open to alternative ideas, hence why I didn't try and finish all of the model creation.

Reply:
Here are dev notes of the things I tried to resolve this issue today, just for reference.

---

* `errorScreen()` private method in the ScreenController is not being called for `/asdf` or `/flow/asdf`. Only for `/flow/ubi/asdf`
* I don't thing `/error` is being called with a `redirect:/error` in the first two that are working
* When we add the status code to ModelAndView, it doesn't even get to `/error`!
- [x] Does using `forward:/error` working?
	* No, it does the same thing as `redirect:`
- [x] Is there a code in the `WebRequest`?
	* No
- [x] What if I add an exception handler to the screen controller?
	* Couldn't get it to work
- [x] There's only `WebRequest` in the error controller, what if there's something I can get from `httpServletRequest`?
	* No, I still can't get the passed in status
- [x] Did the Readme change happen?
	* Yep, needed to change it
- [x] Move file out of `controller` folder since `ScreenController` doesn't work from there
- [x] I should move `error` and `devError` into their own folder in `resources/templates`
- So using `redirect:/error` doesn't work and is the cause of swallowing up the `404` status
- I either need a better way to to move from the `ScreenController` to the error controller endpoint or I need to do the same work and populate `BINDING_ERRORS, EXCEPTION, MESSAGE, STACKTRACE`  in `errorScreen()`

---

Reply:
Now by throwing an exception, we can redirect to our `/error` endpoint with the right status codes being sent!